### PR TITLE
Docker files for CentOS 8 build for 4.1.x

### PR DESCRIPTION
**Pull request by hbeatty**
I have been working on getting TC running on CentOS 8. I thought some others might be interested in these files.

### CODE REVIEW

**Comment by rawlinp** [comment]
Should this be targeted towards the master branch instead of 4.1.x?

File: infrastructure/docker/build/Dockerfile-traffic_monitor-el8 [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		gcc \
+		git \
+		rpm-build && \
+	yum -y clean all
+
+RUN curl -LO https://dl.google.com/go/go1.14.2.linux-amd64.tar.gz && \
+    tar -C /usr/local -xvzf go1.14.2.linux-amd64.tar.gz && \
+    ln -s /usr/local/go/bin/go /usr/bin/go && \
+    rm go1.14.2.linux-amd64.tar.gz
+
```
Comment (jhg03a) [inline infrastructure/docker/build/Dockerfile-traffic_monitor-el8:46]: There were some recent modifications to upgrade to Golang 15

File: infrastructure/docker/build/Dockerfile-traffic_monitor-el7 [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		# gcc is necessary in case -gcflags 'all=-N -l' is passed to go build
+		gcc \
+		git \
+		rpm-build && \
+	yum -y clean all
+
+RUN curl -LO https://dl.google.com/go/go1.14.2.linux-amd64.tar.gz && \
+    tar -C /usr/local -xvzf go1.14.2.linux-amd64.tar.gz && \
+    ln -s /usr/local/go/bin/go /usr/bin/go && \
+    rm go1.14.2.linux-amd64.tar.gz
```
Comment (jhg03a) [inline infrastructure/docker/build/Dockerfile-traffic_monitor-el7:44]: There were some recent modifications to upgrade to Golang 15

File: infrastructure/docker/build/Dockerfile-traffic_ops-el7 [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		openssl-devel \
+		perl-ExtUtils-MakeMaker \
+		rpm-build \
+		tar && \
+	yum -y clean all
+
+RUN curl -LO https://dl.google.com/go/go1.14.2.linux-amd64.tar.gz && \
+    tar -C /usr/local -xvzf go1.14.2.linux-amd64.tar.gz && \
+    ln -s /usr/local/go/bin/go /usr/bin/go && \
+    rm go1.14.2.linux-amd64.tar.gz
```
Comment (jhg03a) [inline infrastructure/docker/build/Dockerfile-traffic_ops-el7:49]: There were some recent modifications to upgrade to Golang 15

File: infrastructure/docker/build/Dockerfile-traffic_ops-el8 [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+		openssl-devel \
+		perl-ExtUtils-MakeMaker \
+		rpm-build \
+		tar && \
+	yum -y clean all
+
+RUN curl -LO https://dl.google.com/go/go1.14.2.linux-amd64.tar.gz && \
+    tar -C /usr/local -xvzf go1.14.2.linux-amd64.tar.gz && \
+    ln -s /usr/local/go/bin/go /usr/bin/go && \
+    rm go1.14.2.linux-amd64.tar.gz
```
Comment (jhg03a) [inline infrastructure/docker/build/Dockerfile-traffic_ops-el8:50]: There were some recent modifications to upgrade to Golang 15

File: infrastructure/docker/build/Dockerfile-traffic_stats-el7 [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+### traffic_stats specific requirements
+RUN	yum -y install \
+		git  \
+		rpm-build && \
+	yum -y clean all
+
+RUN curl -LO https://dl.google.com/go/go1.14.2.linux-amd64.tar.gz && \
+    tar -C /usr/local -xvzf go1.14.2.linux-amd64.tar.gz && \
+    ln -s /usr/local/go/bin/go /usr/bin/go && \
+    rm go1.14.2.linux-amd64.tar.gz
```
Comment (jhg03a) [inline infrastructure/docker/build/Dockerfile-traffic_stats-el7:42]: There were some recent modifications to upgrade to Golang 15

File: infrastructure/docker/build/Dockerfile-traffic_stats-el8 [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+### traffic_stats specific requirements
+RUN	yum -y install \
+		git  \
+		rpm-build && \
+	yum -y clean all
+
+RUN curl -LO https://dl.google.com/go/go1.14.2.linux-amd64.tar.gz && \
+    tar -C /usr/local -xvzf go1.14.2.linux-amd64.tar.gz && \
+    ln -s /usr/local/go/bin/go /usr/bin/go && \
+    rm go1.14.2.linux-amd64.tar.gz
```
Comment (jhg03a) [inline infrastructure/docker/build/Dockerfile-traffic_stats-el8:43]: There were some recent modifications to upgrade to Golang 15

File: infrastructure/docker/build/docker-compose-combo.yml [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    build:
+      dockerfile: infrastructure/docker/build/Dockerfile-source
+      context: ../../..
+    volumes:
+      - ../../..:/trafficcontrol:z
+
+  traffic_monitor_build:
+    image: traffic_monitor_builder
+    build:
+      dockerfile: infrastructure/docker/build/Dockerfile-traffic_monitor-${COS_VERSION}
```
Comment (jhg03a) [inline infrastructure/docker/build/docker-compose-combo.yml:32]: Might be a good idea to have a default `COS_VERSION` specified for 7.  Docker variable intepolation should support this.

**Review by jhg03a — CHANGES_REQUESTED** [review]
Overall looks good, the Golang version and COS_VERSION defaults are the main things I see.  I do wonder though if/what the impact is on the pkg script though since it's still the primary means of building.  Before it gets merged, we'd need to be sure existing calls to pkg work and an arg and environment var can switch COS_VERSION too.  It may also be worth a separate discussion about increasing our testing images + CIAB to support Centos8 as well.  As long as Centos7 is the default and pkg works the same though, I'm ok with that being completely separate.

File: infrastructure/docker/build/Dockerfile-traffic_monitor-el7 [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+###
+
+ADD infrastructure/docker/build/clean_build.sh /
```
Comment (rawlinp) [inline infrastructure/docker/build/Dockerfile-traffic_monitor-el7:48]: On master this script was moved to build/clean_build.sh

**Pull request closed without merging**